### PR TITLE
Mz/update toolchain

### PR DESCRIPTION
**Pull request by mayeul-zama**

### CODE REVIEW

File: tfhe/examples/fhe_strings/server_key/pattern/mod.rs [hunk 1]
```diff
@@ -61,7 +61,7 @@ impl ServerKey {
         str: &'a FheString,
         pat: &'a FheString,
         null: Option<&'a FheAsciiChar>,
-    ) -> (CharIter, CharIter, Range<usize>) {
+    ) -> (CharIter<'a>, CharIter<'a>, Range<usize>) {
```
Comment (IceTDrinker) [inline tfhe/examples/fhe_strings/server_key/pattern/mod.rs:64]: 🧐 that's new

File: tfhe/src/core_crypto/commons/traits/contiguous_entity_container.rs [hunk 2]
```diff
@@ -254,7 +254,7 @@ pub trait ContiguousEntityContainer: AsRef<[Self::Element]> {
     fn par_chunks<'this>(
         &'this self,
         chunk_size: usize,
-    ) -> ParallelChunksWrappingLendingIterator<'_, Self::Element, Self::SelfView<'_>>
+    ) -> ParallelChunksWrappingLendingIterator<'this, Self::Element, Self::SelfView<'this>>
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/commons/traits/contiguous_entity_container.rs:257]: do you have the lint that this is triggered by ? It's weird the inference is not working here

**Review by IceTDrinker — COMMENTED** [review]
just a few questions (maybe some rust regression on lifetime inference)

File: tfhe/src/core_crypto/commons/traits/contiguous_entity_container.rs (hunk 2 above)
Comment (mayeul-zama) [reply tfhe/src/core_crypto/commons/traits/contiguous_entity_container.rs:257]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/builtin/static.ELIDED_NAMED_LIFETIMES.html

File: tfhe/examples/fhe_strings/server_key/pattern/mod.rs (hunk 1 above)
Comment (mayeul-zama) [reply tfhe/examples/fhe_strings/server_key/pattern/mod.rs:64]: Yes

File: tfhe/src/core_crypto/commons/traits/contiguous_entity_container.rs (hunk 2 above)
Comment (IceTDrinker) [reply tfhe/src/core_crypto/commons/traits/contiguous_entity_container.rs:257]: thanks that's very useful, so essentially the lifetime elision is not exactly "find the lifetime yourself"

**Review by IceTDrinker — APPROVED** [review]

**Pull request merged**